### ISSUE TITLE
Can't restart multiple devices since Android Studio 4.0 canary 3

### ISSUE THREAD

**Issue by Pitel** [pbreault/adb-idea#114]
```
java.lang.NoSuchMethodError: org.jetbrains.android.facet.AndroidFacetConfiguration.getAndroidTarget()Lcom/android/sdklib/IAndroidTarget;
	at com.developerphil.adbidea.ui.DeviceChooserDialog.<init>(DeviceChooserDialog.java:40)
	at com.developerphil.adbidea.adb.DeviceResultFetcher.showDeviceChooserDialog(DeviceResultFetcher.kt:59)
	at com.developerphil.adbidea.adb.DeviceResultFetcher.fetch(DeviceResultFetcher.kt:36)
	at com.developerphil.adbidea.adb.AdbFacade.executeOnDevice(AdbFacade.java:73)
	at com.developerphil.adbidea.adb.AdbFacade.restartDefaultActivity(AdbFacade.java:49)
	at com.developerphil.adbidea.action.RestartAction.actionPerformed(RestartAction.java:10)
	at com.developerphil.adbidea.action.AdbAction.actionPerformed(AdbAction.java:13)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:265)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:282)
	at com.intellij.ide.actions.GotoActionAction.lambda$performAction$7(GotoActionAction.java:335)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:82)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransaction$1(TransactionGuardImpl.java:106)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:115)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransactionLater$4(TransactionGuardImpl.java:268)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:433)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:416)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:824)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:412)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:704)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:411)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

They (Studio devs) seem to break it every other release :(

**Comment by AllanWang** [comment]
Related to #113 

Seems like there's a [beta repo](https://android.googlesource.com/platform/tools/adt/idea/+/refs/heads/mirror-goog-studio-master-dev/android/src/org/jetbrains/android/facet) for adt with a completely separate API...

I'm looking into the [plugin verifier](https://github.com/JetBrains/intellij-plugin-verifier) to see if there's a way to figure out what the actual API is. The previous issue was fixed by removing usage, but I don't think that is possible for every situation. I'm still looking for how I can get manifest info in my own plugin

**Comment by eric-labelle** [comment]
Have you had any progress with this issue?

**Comment by gabin8** [comment]
Doesn't work for AS 4.0 Beta 2

**Comment by Pitel** [comment]
Fixed in 1.6.0. Tested with latest Android Studio Android Studio 4.1 Canary 8.
